packages/block-library/src/navigation/edit/index.js

```diff
@@ -513,6 +564,9 @@ function Navigation( {
 								onSelect={ ( { id } ) => {
 									setRef( id );
 								} }
+								onSelectClassic={ ( classicMenu ) => {
+									convert( classicMenu.id, classicMenu.name );
```

Comment:
This kicks of the async process but is non blocking and allows the UI of the block to update independently based on the conversion status provided by the hook.